lib/src/components/NxCombobox/NxCombobox.tsx

```diff
@@ -87,29 +87,31 @@ function NxComboboxRender<T extends string | number | DataItem<string | number,
         [alertDropdownId]: showAlert
       }),
       TextInput = filterInput ? NxFilterInput : NxTextInput,
-      filterInputProps = filterInput === 'search' ? { searchIcon: true } : null;
+      filterInputProps = filterInput === 'search' ? { searchIcon: true } : null,
+      [pageIsFocused, setPageIsFocused] = useState(true);
 
   // There is a requirement that when there is an error querying the data, if the user navigates away from
   // the component and then comes back to it the search should be retried automatically
   function handleComponentFocus(evt: FocusEvent<HTMLDivElement>) {
-    setInputIsFocused(true);
 
+    setInputIsFocused(true);
     if (loadError) {
       // check if this is focus coming into the component from somewhere else on the page, not just moving between
       // children of this component and not from focus coming into the browser from some other window
-      const comingFromOutsidePage = evt.relatedTarget === null,
+      const comingFromOutsidePage = pageIsFocused === false,
```

Comment:
Just wondering if it would be simpler if we remove pageIsFocused state
and just have `!document.hasFocus()`

Something like:
```suggestion
      const comingFromOutsidePage = !document.hasFocus(),
```

Reply:
> Just wondering if it would be simpler if we remove `pageIsFocused` state entirely and just have `!document.hasFocus()` here
> 

So I just tested this, which doesn't work unfortunately. I assume it's because the second you click back into the page, the document now has focus and updates immediately (instead of the state variable which is updated after the fact), and doSearch is called again.

Reply:
> > Just wondering if it would be simpler if we remove `pageIsFocused` state entirely and just have `!document.hasFocus()` here
> 
> So I just tested this, which doesn't work unfortunately. I assume it's because the second you click back into the page, the document now has focus and updates immediately (instead of the state variable which is updated after the fact), and doSearch is called again.

Ah I see